scrapy/utils/testproc.py

```diff
@@ -2,18 +2,27 @@
 
 import os
 import sys
+import warnings
 from typing import TYPE_CHECKING, cast
 
 from twisted.internet.defer import Deferred
 from twisted.internet.error import ProcessTerminated
 from twisted.internet.protocol import ProcessProtocol
 
+from scrapy.exceptions import ScrapyDeprecationWarning
+
 if TYPE_CHECKING:
     from collections.abc import Iterable
 
     from twisted.python.failure import Failure
 
 
+warnings.warn(
+    "The scrapy.utils.testproc module is deprecated.",
+    ScrapyDeprecationWarning,
+)
```

Comment:
[Found 1 use in scrapy-poet](https://github.com/scrapinghub/scrapy-poet/blob/703290b1f7854e2837a82a02bea01a537a033795/tests/test_middleware.py#L15). Not sure if it changes anything, through.

Reply:
For the record, found GitHub code search quite powerful:

> [/(?-i)\btestproc\b/ AND ( org:scrapy OR org:scrapy-plugins  OR org:zytedata OR org:scrapinghub ) AND NOT repo:scrapy/scrapy](https://github.com/search?q=%2F%28%3F-i%29%5Cbtestproc%5Cb%2F+AND+%28+org%3Ascrapy+OR+org%3Ascrapy-plugins++OR+org%3Azytedata+OR+org%3Ascrapinghub+%29+AND+NOT+repo%3Ascrapy%2Fscrapy+&type=code)

Reply:
It indeed looks useful to that scrapy-poet test but most of the `testproc.py` code is generic, only 2 or so lines are specific to Scrapy. So I don't think Scrapy is a good place for shipping it.